test/gtest/uct/test_md.cc

```diff
@@ -704,14 +707,32 @@ UCS_TEST_SKIP_COND_P(test_md, alloc_advise,
     EXPECT_TRUE(address != NULL);
     EXPECT_TRUE(mem.memh != UCT_MEM_HANDLE_NULL);
 
-    status = uct_md_mem_advise(md(), mem.memh, (char *)address + 7,
-                               32 * UCS_KBYTE, UCT_MADV_WILLNEED);
+    status = uct_md_mem_advise(md(), mem.memh, (char*)address + 7, advise_size,
+                               UCT_MADV_WILLNEED);
     EXPECT_UCS_OK(status);
 
     memset(address, 0xBB, size);
     uct_mem_free(&mem);
 }
 
+UCS_TEST_SKIP_COND_P(test_md, alloc_advise,
+                     !check_caps(UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_ADVISE) ||
+                             !has_enough_memory(128 * UCS_MBYTE))
+{
+    test_md_alloc_advise(128 * UCS_MBYTE, 32 * UCS_KBYTE);
+}
+
+UCS_TEST_SKIP_COND_P(test_md, alloc_advise_limited_memory_size,
+                     !check_caps(UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_ADVISE) ||
+                             !has_enough_memory(4 * UCS_KBYTE))
+{
+    size_t max_size    = md_attr().max_alloc;
+    size_t alloc_size  = max_size / 2;
+    size_t advise_size = alloc_size / (4 * UCS_KBYTE);
```

Comment:
Something is weird in this calculation. When you divide "size" by "size", you get a plain number, w/o units..